### PR TITLE
Add cache thread pool to DB calls to improve performance

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
This introduces a static Cache Thread Pool to the SQL module usages and marks the SQL functions as asynchronous calls.
This improves the overall performance when using SQL queries.

The second iteration will include fine-tuning of the thread pool such as introducing a thread pool maxed at the max connections to the DB.

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/120

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/sql/datasource/SQLWorkerThreadPool.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * Class to hold the static cachedThreadPool for SQL.
+ */
+public class SQLWorkerThreadPool {
+
+    public static final ExecutorService SQL_EXECUTOR_SERVICE = Executors.newCachedThreadPool(new SQLThreadFactory());
```
Comment (gimantha) [inline native/src/main/java/io/ballerina/stdlib/sql/datasource/SQLWorkerThreadPool.java:30]: Should we not limit the max threads and tasks in the thread pool? or are we going to let it handle it all itself?

File: native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java [hunk 2]
```diff
@@ -140,9 +175,30 @@ public static Object nativeBatchExecute(BObject client, BArray paramSQLStrings,
...
+                        nativeBatchExecuteExecutable(client, paramSQLStrings, statementParameterProcessor, generateKeys,
+                                false, null);
+                balFuture.complete(resultStream);
+            });
+        } else {
+            return nativeBatchExecuteExecutable(client, paramSQLStrings, statementParameterProcessor, generateKeys,
+                    true, trxResourceManager);
+        }
+        return null;
+
```
Comment (kalaiyarasiganeshalingam) [inline native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java:195]:
```suggestion
```

Comment (niveathika) [reply native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java]: Fixed in d7545f4

File: native/src/main/java/io/ballerina/stdlib/sql/datasource/SQLWorkerThreadPool.java (hunk 1 above)
Comment (niveathika) [reply native/src/main/java/io/ballerina/stdlib/sql/datasource/SQLWorkerThreadPool.java:30]: This is just for the first iteration, We can fine-tune this in the coming iterations

**Comment by niveathika** [comment]
The failure test cases will be fixed once the dependency is updated to https://github.com/ballerina-platform/ballerina-lang/pull/32075

File: ballerina/tests/local-transaction-test.bal [hunk 3]
```diff
@@ -479,6 +479,116 @@ isolated function testLocalTransactionSuccessWithFailedHelper(string status,Mock
...
+    ParameterizedQuery[] sqlQueries =
+        from var row in data
+        select `INSERT INTO Customers (firstName,lastName,registrationID,creditLimit,country)
+                                VALUES (${row.firstName}, ${row.lastName}, ${row.registrationID}, ${row.creditLimit},
+                                ${row.country})`;
+    retry<SQLDefaultRetryManager>(1) transaction {
+        var res = check dbClient->batchExecute(sqlQueries);
+        transInfo = transactions:info();
+        var commitResult = commit;
+        if(commitResult is ()){
```
Comment (daneshk) [inline ballerina/tests/local-transaction-test.bal:504]:
```suggestion
        if commitResult is () {
```

Comment (niveathika) [reply ballerina/tests/local-transaction-test.bal]: Fixed in 209cb86

**Review by kalaiyarasiganeshalingam — APPROVED** [review]

**Pull request merged**